plugins/parsers/bindata/parser_test.go

```diff
@@ -0,0 +1,574 @@
+	{Name: "fieldFoo", Type: "bool"},
+	{Name: "fieldBool1", Type: "bool"},
+	{Name: "fieldUint8", Type: "uint8"},
+	{Name: "fieldInt8", Type: "int8"},
+	{Name: "fieldUint16", Type: "uint16"},
+	{Name: "fieldInt16", Type: "int16"},
+	{Name: "fieldUint32", Type: "uint32"},
+	{Name: "fieldInt32", Type: "int32"},
+	{Name: "fieldUint64", Type: "uint64"},
+	{Name: "fieldInt64", Type: "int64"},
+	{Name: "fieldFloat32", Type: "float32"},
+	{Name: "fieldFloat64", Type: "float64"},
+	{Name: "fieldFoo", Type: "string", Size: 48},
+	{Name: "time", Type: "int32"},
+}
+var defaultTags = map[string]string{
+	"tag0": "value0",
+	"tag1": "value1",
+	"tag2": "value2",
+	"tag3": "value3",
+}
+
+var binaryDataBigEndian = []byte{
+	0x00,
+	0x01,
+	0x02,
+	0xFE,
+	0x00, 0x04,
+	0xFF, 0xFC,
+	0x00, 0x00, 0x00, 0x06,
+	0xFF, 0xFF, 0xFF, 0xFA,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08,
+	0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xF8,
+	0x41, 0x29, 0x00, 0x00,
+	0x40, 0x26, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47,
+	0x48, 0x49, 0x4A, 0x4B, 0x4C, 0x4D, 0x4E, 0x4F,
+	0x50, 0x51, 0x52, 0x53,
+	0x5D, 0xA8, 0x6C, 0x4C,
+}
+
+var binaryDataLittleEndian = []byte{
+	0x00,
+	0x01,
+	0x02,
+	0xFE,
+	0x04, 0x00,
+	0xFC, 0xFF,
+	0x06, 0x00, 0x00, 0x00,
+	0xFA, 0xFF, 0xFF, 0xFF,
+	0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0xF8, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+	0x00, 0x00, 0x29, 0x41,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x26, 0x40,
+	0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47,
+	0x48, 0x49, 0x4A, 0x4B, 0x4C, 0x4D, 0x4E, 0x4F,
+	0x50, 0x51, 0x52, 0x53,
+	0x4C, 0x6C, 0xA8, 0x5D,
+}
+
+var binaryDataStringUTF8 = []byte{
+	0x00,
+	0x01,
+	0x02,
+	0xFE,
+	0x04, 0x00,
+	0xFC, 0xFF,
+	0x06, 0x00, 0x00, 0x00,
+	0xFA, 0xFF, 0xFF, 0xFF,
+	0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0xF8, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+	0x00, 0x00, 0x29, 0x41,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x26, 0x40,
+	0x32, 0x48, 0xe2, 0x82, 0x82, 0x20, 0x2b, 0x20,
+	0x4f, 0xe2, 0x82, 0x82, 0x20, 0xe2, 0x87, 0x8c,
+	0x20, 0x32, 0x48, 0xe2, 0x82, 0x82, 0x4f, 0x2c,
+	0x20, 0x52, 0x20, 0x3d, 0x20, 0x34, 0x2e, 0x37,
+	0x20, 0x6b, 0xce, 0xa9, 0x2c, 0x20, 0xe2, 0x8c,
+	0x80, 0x20, 0x32, 0x30, 0x30, 0x20, 0x6d, 0x6d,
+	0x4C, 0x6C, 0xA8, 0x5D,
+}
+
+var binaryDataBigEndianTimeS = []byte{
+	0x01, 0x6D, 0xD9, 0xE7,
+}
+
+var binaryDataBigEndianTimeXs = []byte{
+	0x00, 0x00, 0x01, 0x6D, 0xD9, 0xE7, 0x0B, 0x17,
+}
+
+func TestBigEndian(t *testing.T) {
+
+	var parser, err = NewBinDataParser(
+		"big_endian",
+		"unix",
+		"be",
+		"utf-8",
+		fields,
+		nil,
+	)
+	require.NoError(t, err)
+	assert.NotNil(t, parser)
+
+	metrics, err := parser.Parse(binaryDataBigEndian)
+	require.NoError(t, err)
+	assert.Len(t, metrics, 1)
+	require.Equal(t, parser.metricName, metrics[0].Name())
+	assert.Equal(t, int64(0x5DA86C4C), metrics[0].Time().Unix())
+	assert.Equal(t, expectedParseResult, metrics[0].Fields())
+}
+
+func TestLittleEndian(t *testing.T) {
```

Comment:
Please also apply the examples above to all following tests.

Reply:
Done